src/commands/dev.ts

```diff
@@ -351,28 +360,34 @@ export async function command(commandOptions: CommandOptions) {
         isBundled: false,
         config,
       });
-      builtFileResult.result = await transformEsmImports(builtFileResult.result, (spec) => {
-        // Try to resolve the specifier to a known URL in the project
-        const resolvedImportUrl = resolveImportSpecifier(spec);
-        if (resolvedImportUrl) {
-          return resolvedImportUrl;
-        }
-        // If that fails, return a placeholder import and attempt to resolve.
-        const packageName = getPackageNameFromSpecifier(spec);
-        const [depManifestLoc] = resolveDependencyManifest(packageName, cwd);
-        const doesPackageExist = !!depManifestLoc;
-        if (doesPackageExist) {
-          reinstallDependencies();
-        } else {
-          missingWebModule = {
-            spec: spec,
-            pkgName: packageName,
-          };
-        }
-        // Return a placeholder while Snowpack goes out and tries to re-install (or warn)
-        // on the missing package.
-        return spec;
-      });
+      builtFileResult.result = await transformFileImports(
+        builtFileResult.result,
+        // This is lame: because routes don't have a file extension, we create
+        // a fake file name with the correct extension, which is the important bit. 
+        'file.' + (srcFileExtensionMapping[ext] || ext),
```

Comment:
If I had a little more time I would have gotten rid of this 🙈 I'll try to clean this bit up before merging, open to suggestions (the only bit of info we need is the file extension / is it JS or HTML?)

Reply:
This part is fine to me because you left a comment on top. To me I think an opportunity for cleanup is above, with lines like:

```js
    if (
      ext === 'js' ||
      srcFileExtensionMapping[ext] === 'js' ||
      ext === 'html' ||
      srcFileExtensionMapping[ext] === 'html'
    ) {
```

It seems like we have logic in several places to determine what a resource “is” (is this JS? HTML? Neither? etc.). Things like that would reduce the mental overhead in this by a lot, I think, because then things like this are explained as feeding into that “identity” of resources

Reply:
100%, i don't like how we rely so heavily on this at the moment. Plus, the difference between `js` and `.js`... 🙈